lock-api/src/main/java/com/palantir/lock/LockService.java

```diff
@@ -87,6 +107,10 @@
      *         {@code token} are read locks (instead of write locks), or if
      *         {@code token} is held by an anonymous lock client
      */
+    @POST
+    @Path("unlock-and-freeze")
```

Comment:
i assume these names are all taken from PG's lock server?

Reply:
If I'm not wrong the original doesn't use a rest-based framework, so there isn't really a precedent. The protobuf lock service had an enum for the various methods, as we would probably do if we did a websocket like impl.

Reply:
Yes, as PG does not have REST APIs, I came up with these names based on the method names.